src/main/java/seedu/smarthomebot/logic/commands/OnCommand.java

```diff
@@ -71,19 +71,21 @@ public CommandResult execute() {
                 return new CommandResult("Invalid Format");
             }
         } catch (ApplianceNotFoundException e) {
-            return new CommandResult(MESSAGE_APPLIANCE_OR_LOCATION_NOT_EXIST);
+            if (locationList.isLocationCreated(argument)) {
+                return new CommandResult("There are no Appliances in \"" + argument + "\".");
+            } else {
+                return new CommandResult(MESSAGE_APPLIANCE_OR_LOCATION_NOT_EXIST);
+            }
         } catch (ParameterFoundException e) {
             return new CommandResult(MESSAGE_NO_PARAMETER_IN_ON_BY_LOCATION);
-        } catch (NoApplianceInLocationException e) {
-            return new CommandResult("There are no Appliances in \"" + argument + "\".");
         }
     }
 
     /**
      * Method to on Appliance by the name.
      */
-    private CommandResult onByApplianceName() throws ApplianceNotFoundException, NoApplianceInLocationException {
-        int toOnApplianceIndex = applianceList.getApplianceIndex(argument, locationList);
+    private CommandResult onByApplianceName() throws ApplianceNotFoundException {
+        int toOnApplianceIndex = applianceList.getApplianceIndex(argument);
```

Comment:
This command is the same as OffCommand. Can we combine it ?

Reply:
What do you mean by combining it?

Reply:
I believe while they are very similar when it comes to code, they serve two different command types and hence should be separated in the spirit of OOP, what do you think?

Reply:
Then i think it is fine